Sources/Contentful/Client.swift

```diff
@@ -614,88 +614,50 @@ extension Client {
-    /**
-     Perform an initial synchronization of the Space this client is constrained to.
-
-     - Parameter syncableTypes: The types that can be synchronized.
-     - Parameter completion: A handler being called on completion of the request.
-
-     - Returns: The data task being used, enables cancellation of requests.
-     */
-    @discardableResult public func initialSync(syncableTypes: SyncSpace.SyncableTypes = .all,
-                                               then completion: @escaping ResultsHandler<SyncSpace>) -> URLSessionDataTask? {
-
-        // Sync currently only works for the master environemnt.
-        guard environmentId == "master" else {
-            completion(Result.error(SDKError.nonMasterEnvironmentsDoNotSupportSync()))
-            return nil
-        }
-
-        let syncCompletion: (Result<SyncSpace>) -> Void = { result in
-            self.finishSync(for: SyncSpace(syncToken: ""),
-                            newestSyncResults: result,
-                            then: completion)
-        }
-        return sync(operation: .initial, syncableTypes: syncableTypes, then: syncCompletion)
-    }
-
-    /**
-     Perform an initial synchronization of the Space this client is constrained to.
-
-     - Parameter matching: Additional options for the synchronization.
-     - Returns: An `Observable` for the resulting SyncSpace.
-     */
-
-    @discardableResult public func initialSync(syncableTypes: SyncSpace.SyncableTypes = .all) -> Observable<Result<SyncSpace>> {
-        let asyncDataTask: AsyncDataTask<SyncSpace.SyncableTypes, SyncSpace> = initialSync(syncableTypes:then:)
-        return toObservable(parameter: syncableTypes, asyncDataTask: asyncDataTask).observable
-    }
 
     /**
-     Perform a subsequent synchronization operation, updating this object with
-     the latest content from Contentful.
+     Perform a synchronization operation, updating the passed in `SyncSpace` object with
+     latest content from Contentful. If the passed in `SyncSpace` is a new empty instance with an empty
+     sync token, a full synchronization will be done.
 
      Calling this will mutate the instance and also return a reference to itself to the completion
      handler in order to allow chaining of operations.
 
-     - Parameter syncSpace: the relevant `SyncSpace` to perform the subsequent sync on.
+     - Parameter syncSpace: the relevant `SyncSpace` to perform the subsequent sync on. Defaults to a new empty instance of sync space.
      - Parameter syncableTypes: The types that can be synchronized.
-
+     
      - Returns: An `Observable` which will be fired when the `SyncSpace` is fully synchronized with Contentful.
      */
-    @discardableResult public func nextSync(for syncSpace: SyncSpace,
-                                            syncableTypes: SyncSpace.SyncableTypes = .all) -> Observable<Result<SyncSpace>> {
+    @discardableResult public func sync(for syncSpace: SyncSpace = SyncSpace(),
```

Comment:
Noob question: does the assignment get executed on every method call or just once? Aka do several initial syncs create different empty objects? 🤔

Reply:
The default argument in this case calls the initializer/constructor for `SyncSpace` and makes a new instance on each call that has the `syncSpace` argument ommited.

So `sync()` would be the equivalent of calling the following

```swift
let emptySyncSpace = SyncSpace()
client.sync(for: emptySyncSpace)
```